contracts/src/protocol/OmniPortal.sol

```diff
@@ -137,13 +137,13 @@ contract OmniPortal is
      * @dev Validate the xcall, emit an XMsg, increment dest chain outXStreamOffset
      */
     function _xcall(uint64 destChainId, address sender, address to, bytes calldata data, uint64 gasLimit) private {
-        require(msg.sender != address(this), "OmniPortal: portal cannot xcall");
-        require(msg.value >= feeFor(destChainId, data, gasLimit), "OmniPortal: insufficient fee");
-        require(gasLimit <= xmsgMaxGasLimit, "OmniPortal: gasLimit too high");
-        require(gasLimit >= xmsgMinGasLimit, "OmniPortal: gasLimit too low");
         require(destChainId != chainId(), "OmniPortal: no same-chain xcall");
         require(destChainId != _BROADCAST_CHAIN_ID, "OmniPortal: no broadcast xcall");
         require(to != _VIRTUAL_PORTAL_ADDRESS, "OmniPortal: no portal xcall");
+        require(gasLimit <= xmsgMaxGasLimit, "OmniPortal: gasLimit too high");
+        require(gasLimit >= xmsgMinGasLimit, "OmniPortal: gasLimit too low");
+        require(msg.sender != address(this), "OmniPortal: portal cannot xcall");
+        require(msg.value >= feeFor(destChainId, data, gasLimit), "OmniPortal: insufficient fee");
```

Comment:
Just changed some require ordering, so we do feeFor last. So we avoid "no fee parameters" errors for cases that should be handled by other statments (like "no same-chain xcall")